dropshot/src/api_description.rs

```diff
@@ -78,7 +78,7 @@ impl<'a, Context: ServerContext> ApiEndpoint<Context> {
             summary: None,
             description: None,
             tags: vec![],
-            paginated: func_parameters.paginated,
+            extension_mode: func_parameters.extension_mode,
```

Comment:
So this means a given endpoint can only be in one mode at a time, right? That makes sense for pagination and websockets, since they're mutually exclusive. I have no idea what other kinds of extensions we might dream up, but is it possible we could want to use more than one at a time?

Reply:
yeah - i suppose if such a thing comes to pass, we could change to be something more sophisticated than a bare enum, but i didn't want to try to anticipate unknown unknowns yet, just represent that pagination and websockets are mutually exclusive extensions